website/addons/owncloud/serializer.py

```diff
@@ -0,0 +1,81 @@
+from website.addons.base.serializer import OAuthAddonSerializer
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.util import api_url_for, web_url_for
+from website.addons.owncloud.utils import ExternalAccountConverter
+
+class OwnCloudSerializer(OAuthAddonSerializer):
```

Comment:
This will require implementing a `credentials_are_valid` method, which may make a trivial request to the connected ownCloud to test for credential validity. Overall, though, it should simplify this class.